OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -127,6 +129,11 @@ public IEnumerable<IRenderable> Render(WorldRenderer wr)
 
 		public bool HasRenderables { get { return renderables.Any(); } }
 
+		public bool CheckForRemoval(Player owner)
+		{
+			return removeFrozenActors.Any(r => r.RemoveActor(actor, owner));
```

Comment:
Please use a loop here for performance reasons - frozen actors are a performance sensitive part of the game.